src/main/java/net/snowflake/client/jdbc/SnowflakeConnectionV1.java

```diff
@@ -500,14 +423,18 @@ public Statement createStatement(int resultSetType, int resultSetConcurrency)
   @Override
   public Statement createStatement(
       int resultSetType, int resultSetConcurrency, int resultSetHoldability) throws SQLException {
-    logger.debug(
-        "Statement createStatement(int resultSetType, "
-            + "int resultSetConcurrency, int resultSetHoldability");
-
-    Statement stmt =
-        new SnowflakeStatementV1(this, resultSetType, resultSetConcurrency, resultSetHoldability);
-    openStatements.add(stmt);
-    return stmt;
+    if (isUsingDefaultConnectionHandler) {
```

Comment:
Same comment as above. I don't understand the rationale for these checks. .  I would just invoke the method on the this.connectionHandler

Reply:
Ah, good question! This is for safety. We'll remove this boolean once we have the other interfaces. Because we're splitting these PRs and don't yet have the other interfaces, we'll need to fail gracefully when this code path is hit.